test/functional/legacy/prompt_buffer_spec.lua

```diff
@@ -245,4 +246,280 @@ describe('prompt buffer', function()
       Leave
       Close]])
   end)
+
+  it('can insert mutli line text', function()
+    source_script()
+    feed('line 1<s-cr>line 2<s-cr>line 3')
+    screen:expect([[
+      cmd: line 1              |
+      line 2                   |
+      line 3^                   |
+      {1:~                        }|
+      {3:[Prompt] [+]             }|
+      other buffer             |
+      {1:~                        }|*3
+      {5:-- INSERT --}             |
+    ]])
+
+    feed('<cr>')
+    -- submiting multiline text works
+    screen:expect([[
+      Result: "line 1          |
+      line 2                   |
+      line 3"                  |
+      cmd: ^                    |
+      {3:[Prompt]                 }|
+      other buffer             |
+      {1:~                        }|*3
+      {5:-- INSERT --}             |
+    ]])
+  end)
+
+  it('can paste multiline text', function()
```

Comment:
also need a test for `nvim_paste()` explicitly, since that is a rather different codepath.